
% Template for PLoS
% Version 3.5 March 2018
%
% % % % % % % % % % % % % % % % % % % % % %
%
% -- IMPORTANT NOTE
%
% This template contains comments intended 
% to minimize problems and delays during our production 
% process. Please follow the template instructions
% whenever possible.
%
% % % % % % % % % % % % % % % % % % % % % % % 
%
% Once your paper is accepted for publication, 
% PLEASE REMOVE ALL TRACKED CHANGES in this file 
% and leave only the final text of your manuscript. 
% PLOS recommends the use of latexdiff to track changes during review, as this will help to maintain a clean tex file.
% Visit https://www.ctan.org/pkg/latexdiff?lang=en for info or contact us at latex@plos.org.
%
%
% There are no restrictions on package use within the LaTeX files except that 
% no packages listed in the template may be deleted.
%
% Please do not include colors or graphics in the text.
%
% The manuscript LaTeX source should be contained within a single file (do not use \input, \externaldocument, or similar commands).
%
% % % % % % % % % % % % % % % % % % % % % % %
%
% -- FIGURES AND TABLES
%
% Please include tables/figure captions directly after the paragraph where they are first cited in the text.
%
% DO NOT INCLUDE GRAPHICS IN YOUR MANUSCRIPT
% - Figures should be uploaded separately from your manuscript file. 
% - Figures generated using LaTeX should be extracted and removed from the PDF before submission. 
% - Figures containing multiple panels/subfigures must be combined into one image file before submission.
% For figure citations, please use "Fig" instead of "Figure".
% See http://journals.plos.org/plosone/s/figures for PLOS figure guidelines.
%
% Tables should be cell-based and may not contain:
% - spacing/line breaks within cells to alter layout or alignment
% - do not nest tabular environments (no tabular environments within tabular environments)
% - no graphics or colored text (cell background color/shading OK)
% See http://journals.plos.org/plosone/s/tables for table guidelines.
%
% For tables that exceed the width of the text column, use the adjustwidth environment as illustrated in the example table in text below.
%
% % % % % % % % % % % % % % % % % % % % % % % %
%
% -- EQUATIONS, MATH SYMBOLS, SUBSCRIPTS, AND SUPERSCRIPTS
%
% IMPORTANT
% Below are a few tips to help format your equations and other special characters according to our specifications. For more tips to help reduce the possibility of formatting errors during conversion, please see our LaTeX guidelines at http://journals.plos.org/plosone/s/latex
%
% For inline equations, please be sure to include all portions of an equation in the math environment.  For example, x$^2$ is incorrect; this should be formatted as $x^2$ (or $\mathrm{x}^2$ if the romanized font is desired).
%
% Do not include text that is not math in the math environment. For example, CO2 should be written as CO\textsubscript{2} instead of CO$_2$.
%
% Please add line breaks to long display equations when possible in order to fit size of the column. 
%
% For inline equations, please do not include punctuation (commas, etc) within the math environment unless this is part of the equation.
%
% When adding superscript or subscripts outside of brackets/braces, please group using {}.  For example, change "[U(D,E,\gamma)]^2" to "{[U(D,E,\gamma)]}^2". 
%
% Do not use \cal for caligraphic font.  Instead, use \mathcal{}
%
% % % % % % % % % % % % % % % % % % % % % % % % 
%
% Please contact latex@plos.org with any questions.
%
% % % % % % % % % % % % % % % % % % % % % % % %

\documentclass[10pt,letterpaper]{article}
\usepackage[top=0.85in,left=2.75in,footskip=0.75in]{geometry}

% amsmath and amssymb packages, useful for mathematical formulas and symbols
\usepackage{amsmath,amssymb}

% Use adjustwidth environment to exceed column width (see example table in text)
\usepackage{changepage}

% Use Unicode characters when possible
\usepackage[utf8x]{inputenc}

% textcomp package and marvosym package for additional characters
\usepackage{textcomp,marvosym}

% cite package, to clean up citations in the main text. Do not remove.
\usepackage{cite}

% Use nameref to cite supporting information files (see Supporting Information section for more info)
\usepackage{nameref,hyperref}

% line numbers
\usepackage[right]{lineno}

% ligatures disabled
\usepackage{microtype}
\DisableLigatures[f]{encoding = *, family = * }

% color can be used to apply background shading to table cells only
\usepackage[table]{xcolor}

% array package and thick rules for tables
\usepackage{array}

% create "+" rule type for thick vertical lines
\newcolumntype{+}{!{\vrule width 2pt}}

% create \thickcline for thick horizontal lines of variable length
\newlength\savedwidth
\newcommand\thickcline[1]{%
  \noalign{\global\savedwidth\arrayrulewidth\global\arrayrulewidth 2pt}%
  \cline{#1}%
  \noalign{\vskip\arrayrulewidth}%
  \noalign{\global\arrayrulewidth\savedwidth}%
}

% \thickhline command for thick horizontal lines that span the table
\newcommand\thickhline{\noalign{\global\savedwidth\arrayrulewidth\global\arrayrulewidth 2pt}%
\hline
\noalign{\global\arrayrulewidth\savedwidth}}


% Remove comment for double spacing
%\usepackage{setspace} 
%\doublespacing

% Text layout
\raggedright
\setlength{\parindent}{0.5cm}
\textwidth 5.25in 
\textheight 8.75in

% Bold the 'Figure #' in the caption and separate it from the title/caption with a period
% Captions will be left justified
\usepackage[aboveskip=1pt,labelfont=bf,labelsep=period,justification=raggedright,singlelinecheck=off]{caption}
\renewcommand{\figurename}{Fig}

% Use the PLoS provided BiBTeX style
\bibliographystyle{plos2015}

% Remove brackets from numbering in List of References
\makeatletter
\renewcommand{\@biblabel}[1]{\quad#1.}
\makeatother



% Header and Footer with logo
\usepackage{lastpage,fancyhdr,graphicx}
\usepackage{epstopdf}
%\pagestyle{myheadings}
\pagestyle{fancy}
\fancyhf{}
%\setlength{\headheight}{27.023pt}
%\lhead{\includegraphics[width=2.0in]{PLOS-submission.eps}}
\rfoot{\thepage/\pageref{LastPage}}
\renewcommand{\headrulewidth}{0pt}
\renewcommand{\footrule}{\hrule height 2pt \vspace{2mm}}
\fancyheadoffset[L]{2.25in}
\fancyfootoffset[L]{2.25in}
\lfoot{\today}

%% Include all macros below

\newcommand{\lorem}{{\bf LOREM}}
\newcommand{\ipsum}{{\bf IPSUM}}

\newcommand{\extendedGNW}{GeneNetWeaverPhos}
\newcommand{\PKTFX}{PhosTF}
% from custom.tex
\newcommand{\TF}{\mathrm{TF}}
\newcommand{\PK}{\mathrm{PK}}
\newcommand{\PP}{\mathrm{PP}}
\newcommand{\KP}{\mathrm{KP}}
\newcommand{\KPTF}{\KP\TF}
\newcommand{\plus}{\textup{\texttt{+}}}
\newcommand{\minus}{\textup{\texttt{-}}}
\newcommand{\WP}{W_\KP}
\newcommand{\WT}{W_\TF}
\newcommand{\pos}{{\multimapdot}}
\renewcommand{\neg}{{\multimap}}
% symbol light edges:
\newcommand{\edge}[2]{d(#1,#2)}
\newcommand{\edgePos}[2]{d(#1,#2,+)}
\newcommand{\edgeNeg}[2]{d(#1,#2,-)}
\newcommand{\edgePhos}[2]{d(#1,#2)}
\newcommand{\edgePosPhos}[2]{d(#1,#2,+)}
\newcommand{\edgeNegPhos}[2]{d(#1,#2,-)}
% from math.tex
\usepackage{textgreek} % greek letters in text
\usepackage{amsmath}
\usepackage{amsfonts}
\usepackage{amssymb}
\usepackage{physics}
\usepackage{siunitx}
\usepackage{gensymb}
\usepackage{mathtools} % \coloneqq
\usepackage{txfonts} %
\sisetup{per-mode=fraction}
% molar unit is added
\DeclareSIUnit\molar{M}
\DeclareMathOperator{\cov}{\text{cov}}
\DeclareMathOperator{\E}{\text{E}}
\DeclareMathOperator{\sign}{sgn}
% transpose version of T
\newcommand{\trans}{\intercal}
% make a good looking tilde. otherwise use \sim or \textasciitilde
\newcommand{\realtilde}{{\raise.17ex\hbox{$\scriptstyle\sim$}}}
% creates raised text in math
\newcommand{\lift}[1]{^{(\text{#1})}}
\DeclareMathOperator*{\argmax}{arg\,max}
\DeclareMathOperator*{\argmin}{arg\,min}



%% END MACROS SECTION


\begin{document}
\vspace*{0.2in}

% Title must be 250 characters or less.
\begin{flushleft}
{\Large
\textbf\newline{Systematic inference of regulation by protein kinases finds surprising level of transcription factor deactivation} % Please use "sentence case" for title and headings (capitalize only the first word in a title (or heading), the first word in a subtitle (or subheading), and any proper nouns).
}
\newline
% Insert author names, affiliations and corresponding author email (do not include titles, positions, or degrees).
\\
Christian Degnbol Madsen\textsuperscript{1,2},
Jotun Hein\textsuperscript{2},
Christopher T. Workman\textsuperscript{1*}
\\
\bigskip
\textbf{1} Department of Biotechnology and Biomedicine, Technical University of Denmark, 2800 Kongens Lyngby, Denmark
\\
\textbf{2} Department of Statistics, University of Oxford, Oxford OX1, UK
\\
\bigskip

% Use the asterisk to denote corresponding authorship and provide email address in note below.
* cwor@dtu.dk

\end{flushleft}
% Please keep the abstract below 300 words
\section*{Abstract}

Gene expression is controlled by pathways of regulatory factors often involving the activity of protein kinases on transcription factor proteins.
Despite this well established mechanism, the number of well described pathways that include the regulatory role of protein kinases on transcription factors is surprisingly small in eukaryotes.
%This has motivated approaches that measure or infer these poorly mapped regulatory interactions. 

To address this, \textit{\PKTFX{}} was developed to infer functional regulatory interactions and pathways in both simulated and real biological networks, based on linear cyclic causal models with latent variables. 
%To address this, we developed a novel inference method \textit{\PKTFX{}}, based on linear cyclic causal models with latent variables, that integrates protein interactions with measurements of gene expression to infer functional regulatory interactions and pathways in both simulated and real biological networks.
\textit{\extendedGNW{}}, an extension of \textit{GeneNetWeaver}, was developed to allow the simulation of perturbations in known networks that included the activity of protein kinases and phosphatases on gene regulation.
%Performance was evaluated on networks and perturbations simulated with an extension of \textit{GeneNetWeaver} that included the activity of protein kinases and phosphatases on regulation, \textit{\extendedGNW{}}.
Over 2000 genome-wide gene expression profiles, where the loss or gain of regulatory genes could be observed to perturb gene regulation, were then used to infer the existence of regulatory interactions, and their mode of regulation in the budding yeast \textit{Saccharomyces cerevisiae}.
%In particular, our approach inferred the existence of regulatory interactions, and their mode of regulation, using gene expression profiles where the loss or gain of regulatory genes could be observed to perturb the output of regulatory networks.

%\PKTFX{} was first tested on networks and perturbations simulated with an extension of \textit{GeneNetWeaver} that included the activity of protein kinases and phosphatases on regulation, \textit{\extendedGNW{}}.
Despite the additional complexity, our inference performed comparably to the best methods that inferred transcription factor regulation assessed in the \textit{DREAM4} challenge on similar simulated networks.
Inference on integrated genome-scale data sets for yeast identified $\boldsymbol\sim$8800 protein kinase-transcription factor interactions and $\boldsymbol\sim$6500 interaction between protein kinases and phosphatases.
%Finally, a regulatory network was inferred for budding yeast containing $\boldsymbol\sim$8800 protein kinase-transcription factor interactions and $\boldsymbol\sim$6500 interaction between protein kinases and phosphatases. 
Both types of regulatory predictions captured statistically significant numbers of known interactions of their type. Surprisingly, kinases and phosphatases regulated transcription factors by a negative mode or regulation (deactivation) in over 70\% of the predictions. 
\mdseries\normalsize

Keywords: regulatory network inference, protein kinases, transcription factors, computational modeling, yeast


% Please keep the Author Summary between 150 and 200 words
% Use first person. PLOS ONE authors please skip this step. 
% Author Summary not valid for PLOS ONE submissions.   
\section*{Author summary}
Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur eget porta erat. Morbi consectetur est vel gravida pretium. Suspendisse ut dui eu ante cursus gravida non sed sem. Nullam sapien tellus, commodo id velit id, eleifend volutpat quam. Phasellus mauris velit, dapibus finibus elementum vel, pulvinar non tellus. Nunc pellentesque pretium diam, quis maximus dolor faucibus id. Nunc convallis sodales ante, ut ullamcorper est egestas vitae. Nam sit amet enim ultrices, ultrices elit pulvinar, volutpat risus.

\linenumbers

% Use "Eq" instead of "Equation" for equation citations.
\section*{Introduction}
% MOTIVATION
Gene regulation is central to a cell's ability to respond and adapt to changes in its environment.
The control of transcription rates are directly regulated by transcription factors~(TFs), and indirectly by chromatin state, cell signalling and other regulatory factors.
%Signals from the environment are often propagated to the nucleus by post-translational modifications of the proteins involved. 
Modulation of TF activity is often achieved through phosphorylation or dephosphorylation by protein kinases~(PKs) or phosphatases~(PPs), and TFs represent one of the most phosphorylated classes of proteins~\pcite{Ptacek2005}.
%The model organism \textit{Saccharomyces cerevisiae}, budding yeast, has been thoroughly mapped for protein interactions. 
%In spite of this, most of its regulatory pathways are not fully understood. Budding yeast has ~5200 verified open reading frames~(ORFs)~\pcite{YeastOverview} and its number of TFs, PKs and PPs has been estimated as 209~\pcite{Hughes2013}, 129 and 30~\pcite{Breitkreutz2010}, respectively.
Regulation by TFs can be mapped from protein-DNA binding experiments, e.g. by chromatin immunoprecipitation~(ChIP) based methods, while protein kinase and phosphatase~(KP) regulation can be inferred from protein-protein binding as measured by yeast two-hybrid, or co-immunoprecipitation and mass spectrometry-based methods. These technologies suffer from false negatives due to the transient nature by which kinases and phosphatases bind their targets, as well as false positives~\pcite{Deane2002}. Online databases containing protein interactions will sometimes report whether the data is collected from low- or high-throughput experiments, or whether they were observed reproducibly in multiple experiments, but information about data quality or functionality is often limited~\pcite{Yeang2004}. 
%Another issue is causality, since detecting a protein binding pair does not inform directionality. 
To infer functional regulatory interactions, one can draw from multiple sources of data, both protein binding data and evidence of regulation from mRNA transcript levels. In particular, when comparing the transcript levels from mutant strains, e.g. gene knock-outs or overexpressions, to their background strains, the output of regulatory pathways can be observed by the resulting changes in mRNA levels. The loss or gain of a TF or KP gene will often generate altered transcript levels that imply functional regulation or a regulatory dependency between the perturbed regulator and the gene with an altered mRNA level~\pcite{ChuaPNAS2006}. Inference of functional regulation can be achieved through the use of these diverse data types as evidenced by a number of approaches used in the DREAM4 challenge~\pcite{dream4}.

% GNW
Methods for the inference of TF-based regulatory networks were evaluated in the DREAM4 challenge~\pcite{dream4}. To do this, mRNA concentrations were simulated with the software GeneNetWeaver from a known network using differential equations describing mRNA and protein concentration gradients~\pcite{GeneNetWeaver}. In this way, all regulators can be deleted or overexpressed (in silico) in turn and new steady-state mRNA output can be generated for each. 
%The best performing methods were ensemble methods drawing strength from the "wisdom of the crowd" by combining multiple simpler methods and often employing heuristics. 
However, GeneNetWeaver does not take phosphorylation or other post-translational modification into account.
The focus of our approach was to extend the inference of TF-based regulatory networks to include the activity of kinases and phosphatases, and to apply this method to the model budding yeast \textit{Saccharomyces cerevisiae}, which has been thoroughly mapped for protein interactions.
%For this purpose, GeneNetWeaver was extended to include KP-interactions such that testing data sets could be generated and the performance of our methods could be evaluated.

% PREVIOUS WORK % EBERHARDT
Efforts to infer regulatory networks have focused on TFs binding to target gene's promoter regions, and are often modeled as directed acyclic graphs~(DAG) of TF nodes interacting with nodes representing target genes.
Applied in this biological context, each node value represents a protein's concentration and each edge the direct regulatory effect, or activity, from node to node. 
Modelling regulation as a DAG has limitations on the extent to which graph edges represent causality and physical interaction, since gene regulation is highly cyclic as target gene products are often regulators themselves. 
The \textit{linear cyclic causal models with latent variables}~(LLC) approach was specifically designed to address inference of causality in cyclic graphs and has been applied to infer TF regulatory networks~\pcite{EberhardtLLC}.

% GOAL
\begin{figure}[H]
    \begin{center}
    \includegraphics[width=\textwidth]{introduction/fig/deletion_effects_v6.pdf}
    \end{center}
    \Caption{Effects of gene deletion on gene expression}{Schematic of gene expression levels for a target gene 'V' relative to wildtype when a gene for either a PK or TF is deleted. All combinations of positive (activating) and negative (repressing) regulation (Pos. or Neg. respectively) are shown. Activating or repressing phosphorylation (Phos.) are indicated with closed or open circles and regulation by TFs (Reg.) are indicated with pointed and flat arrowheads.}
    \label{fig:deletion_effect}
\end{figure}

Methods have also been proposed that combine multiple likelihood functions for numerous types of evidence~\pcite{Yeang2004}. In such cases, maximum likelihood ratios can be calculated for each potential regulatory interaction (edge) by describing the likelihood ratios through factor graphs.
Inferring regulation from KPs is much more challenging since they do not regulate mRNA production rates directly, but rather modulate protein activity of other potential regulators. Recent studies utilizing mass spectrometry based proteomics or phosphoproteomics have investigated the activity of KPs in knockout studies~\citep{Goncalves2017,zelezniak2018}. Though these studies endeavour to infer KP activity, they do so relative to the effects on metabolic processes induced by environmental perturbations, thus associating KPs to cellular responses rather than specific regulators.

%In this paper we extent the LLC method in \PKTFX{} in an effort to explore the evidence of indirect regulation caused by kinase and phosphatase activity on TFs.
In this paper we {\color{red} have developed \PKTFX{}, which builds on the LLC method. \PKTFX{} allows for inference of} indirect regulation caused by kinase and phosphatase activity on TFs.
Although KP regulation is difficult to infer from any single knockout, their activity can be inferred in combination with TF knockouts as illustrated in~\autoref{fig:deletion_effect}. We extended GeneNetWeaver simulations in \extendedGNW{} to include regulation by phosphorylation, and describe a new method to infer gene regulatory networks based on a corpus of protein interaction data and a large compendium of knockout and overexpression transcription profiles. 









Lorem ipsum dolor sit~\cite{bib1} amet, consectetur adipiscing elit. Curabitur eget porta erat. Morbi consectetur est vel gravida pretium. Suspendisse ut dui eu ante cursus gravida non sed sem. Nullam Eq~(\ref{eq:schemeP}) sapien tellus, commodo id velit id, eleifend volutpat quam. Phasellus mauris velit, dapibus finibus elementum vel, pulvinar non tellus. Nunc pellentesque pretium diam, quis maximus dolor faucibus id.~\cite{bib2} Nunc convallis sodales ante, ut ullamcorper est egestas vitae. Nam sit amet enim ultrices, ultrices elit pulvinar, volutpat risus.

\begin{eqnarray}
\label{eq:schemeP}
	\mathrm{P_Y} = \underbrace{H(Y_n) - H(Y_n|\mathbf{V}^{Y}_{n})}_{S_Y} + \underbrace{H(Y_n|\mathbf{V}^{Y}_{n})- H(Y_n|\mathbf{V}^{X,Y}_{n})}_{T_{X\rightarrow Y}},
\end{eqnarray}

\section*{Materials and methods}
\subsection*{Etiam eget sapien nibh}

% For figure citations, please use "Fig" instead of "Figure".
Nulla mi mi, Fig~\ref{fig1} venenatis sed ipsum varius, volutpat euismod diam. Proin rutrum vel massa non gravida. Quisque tempor sem et dignissim rutrum. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi at justo vitae nulla elementum commodo eu id massa. In vitae diam ac augue semper tincidunt eu ut eros. Fusce fringilla erat porttitor lectus cursus, \nameref{S1_Video} vel sagittis arcu lobortis. Aliquam in enim semper, aliquam massa id, cursus neque. Praesent faucibus semper libero.

% Place figure captions after the first paragraph in which they are cited.
\begin{figure}[!h]
\caption{{\bf Bold the figure title.}
Figure caption text here, please use this space for the figure panel descriptions instead of using subfigure commands. A: Lorem ipsum dolor sit amet. B: Consectetur adipiscing elit.}
\label{fig1}
\end{figure}

% Results and Discussion can be combined.
\section*{Results}
Nulla mi mi, venenatis sed ipsum varius, Table~\ref{table1} volutpat euismod diam. Proin rutrum vel massa non gravida. Quisque tempor sem et dignissim rutrum. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi at justo vitae nulla elementum commodo eu id massa. In vitae diam ac augue semper tincidunt eu ut eros. Fusce fringilla erat porttitor lectus cursus, vel sagittis arcu lobortis. Aliquam in enim semper, aliquam massa id, cursus neque. Praesent faucibus semper libero.

% Place tables after the first paragraph in which they are cited.
\begin{table}[!ht]
\begin{adjustwidth}{-2.25in}{0in} % Comment out/remove adjustwidth environment if table fits in text column.
\centering
\caption{
{\bf Table caption Nulla mi mi, venenatis sed ipsum varius, volutpat euismod diam.}}
\begin{tabular}{|l+l|l|l|l|l|l|l|}
\hline
\multicolumn{4}{|l|}{\bf Heading1} & \multicolumn{4}{|l|}{\bf Heading2}\\ \thickhline
$cell1 row1$ & cell2 row 1 & cell3 row 1 & cell4 row 1 & cell5 row 1 & cell6 row 1 & cell7 row 1 & cell8 row 1\\ \hline
$cell1 row2$ & cell2 row 2 & cell3 row 2 & cell4 row 2 & cell5 row 2 & cell6 row 2 & cell7 row 2 & cell8 row 2\\ \hline
$cell1 row3$ & cell2 row 3 & cell3 row 3 & cell4 row 3 & cell5 row 3 & cell6 row 3 & cell7 row 3 & cell8 row 3\\ \hline
\end{tabular}
\begin{flushleft} Table notes Phasellus venenatis, tortor nec vestibulum mattis, massa tortor interdum felis, nec pellentesque metus tortor nec nisl. Ut ornare mauris tellus, vel dapibus arcu suscipit sed.
\end{flushleft}
\label{table1}
\end{adjustwidth}
\end{table}


%PLOS does not support heading levels beyond the 3rd (no 4th level headings).
\subsection*{\lorem\ and \ipsum\ nunc blandit a tortor}
\subsubsection*{3rd level heading} 
Maecenas convallis mauris sit amet sem ultrices gravida. Etiam eget sapien nibh. Sed ac ipsum eget enim egestas ullamcorper nec euismod ligula. Curabitur fringilla pulvinar lectus consectetur pellentesque. Quisque augue sem, tincidunt sit amet feugiat eget, ullamcorper sed velit. Sed non aliquet felis. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris commodo justo ac dui pretium imperdiet. Sed suscipit iaculis mi at feugiat. 

\begin{enumerate}
	\item{react}
	\item{diffuse free particles}
	\item{increment time by dt and go to 1}
\end{enumerate}


\subsection*{Sed ac quam id nisi malesuada congue}

Nulla mi mi, venenatis sed ipsum varius, volutpat euismod diam. Proin rutrum vel massa non gravida. Quisque tempor sem et dignissim rutrum. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi at justo vitae nulla elementum commodo eu id massa. In vitae diam ac augue semper tincidunt eu ut eros. Fusce fringilla erat porttitor lectus cursus, vel sagittis arcu lobortis. Aliquam in enim semper, aliquam massa id, cursus neque. Praesent faucibus semper libero.

\begin{itemize}
	\item First bulleted item.
	\item Second bulleted item.
	\item Third bulleted item.
\end{itemize}

\section*{Discussion}
Nulla mi mi, venenatis sed ipsum varius, Table~\ref{table1} volutpat euismod diam. Proin rutrum vel massa non gravida. Quisque tempor sem et dignissim rutrum. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi at justo vitae nulla elementum commodo eu id massa. In vitae diam ac augue semper tincidunt eu ut eros. Fusce fringilla erat porttitor lectus cursus, vel sagittis arcu lobortis. Aliquam in enim semper, aliquam massa id, cursus neque. Praesent faucibus semper libero~\cite{bib3}.

\section*{Conclusion}

CO\textsubscript{2} Maecenas convallis mauris sit amet sem ultrices gravida. Etiam eget sapien nibh. Sed ac ipsum eget enim egestas ullamcorper nec euismod ligula. Curabitur fringilla pulvinar lectus consectetur pellentesque. Quisque augue sem, tincidunt sit amet feugiat eget, ullamcorper sed velit. 

Sed non aliquet felis. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris commodo justo ac dui pretium imperdiet. Sed suscipit iaculis mi at feugiat. Ut neque ipsum, luctus id lacus ut, laoreet scelerisque urna. Phasellus venenatis, tortor nec vestibulum mattis, massa tortor interdum felis, nec pellentesque metus tortor nec nisl. Ut ornare mauris tellus, vel dapibus arcu suscipit sed. Nam condimentum sem eget mollis euismod. Nullam dui urna, gravida venenatis dui et, tincidunt sodales ex. Nunc est dui, sodales sed mauris nec, auctor sagittis leo. Aliquam tincidunt, ex in facilisis elementum, libero lectus luctus est, non vulputate nisl augue at dolor. For more information, see \nameref{S1_Appendix}.

\section*{Supporting information}

% Include only the SI item label in the paragraph heading. Use the \nameref{label} command to cite SI items in the text.
\paragraph*{S1 Fig.}
\label{S1_Fig}
{\bf Bold the title sentence.} Add descriptive text after the title of the item (optional).

\paragraph*{S2 Fig.}
\label{S2_Fig}
{\bf Lorem ipsum.} Maecenas convallis mauris sit amet sem ultrices gravida. Etiam eget sapien nibh. Sed ac ipsum eget enim egestas ullamcorper nec euismod ligula. Curabitur fringilla pulvinar lectus consectetur pellentesque.

\paragraph*{S1 File.}
\label{S1_File}
{\bf Lorem ipsum.}  Maecenas convallis mauris sit amet sem ultrices gravida. Etiam eget sapien nibh. Sed ac ipsum eget enim egestas ullamcorper nec euismod ligula. Curabitur fringilla pulvinar lectus consectetur pellentesque.

\paragraph*{S1 Video.}
\label{S1_Video}
{\bf Lorem ipsum.}  Maecenas convallis mauris sit amet sem ultrices gravida. Etiam eget sapien nibh. Sed ac ipsum eget enim egestas ullamcorper nec euismod ligula. Curabitur fringilla pulvinar lectus consectetur pellentesque.

\paragraph*{S1 Appendix.}
\label{S1_Appendix}
{\bf Lorem ipsum.} Maecenas convallis mauris sit amet sem ultrices gravida. Etiam eget sapien nibh. Sed ac ipsum eget enim egestas ullamcorper nec euismod ligula. Curabitur fringilla pulvinar lectus consectetur pellentesque.

\paragraph*{S1 Table.}
\label{S1_Table}
{\bf Lorem ipsum.} Maecenas convallis mauris sit amet sem ultrices gravida. Etiam eget sapien nibh. Sed ac ipsum eget enim egestas ullamcorper nec euismod ligula. Curabitur fringilla pulvinar lectus consectetur pellentesque.

\section*{Acknowledgments}
Cras egestas velit mauris, eu mollis turpis pellentesque sit amet. Interdum et malesuada fames ac ante ipsum primis in faucibus. Nam id pretium nisi. Sed ac quam id nisi malesuada congue. Sed interdum aliquet augue, at pellentesque quam rhoncus vitae.

\nolinenumbers

% Either type in your references using
% \begin{thebibliography}{}
% \bibitem{}
% Text
% \end{thebibliography}
%
% or
%
% Compile your BiBTeX database using our plos2015.bst
% style file and paste the contents of your .bbl file
% here. See http://journals.plos.org/plosone/s/latex for 
% step-by-step instructions.
% 
\begin{thebibliography}{10}

\bibitem{bib1}
Conant GC, Wolfe KH.
\newblock {{T}urning a hobby into a job: how duplicated genes find new
  functions}.
\newblock Nat Rev Genet. 2008 Dec;9(12):938--950.

\bibitem{bib2}
Ohno S.
\newblock Evolution by gene duplication.
\newblock London: George Alien \& Unwin Ltd. Berlin, Heidelberg and New York:
  Springer-Verlag.; 1970.

\bibitem{bib3}
Magwire MM, Bayer F, Webster CL, Cao C, Jiggins FM.
\newblock {{S}uccessive increases in the resistance of {D}rosophila to viral
  infection through a transposon insertion followed by a {D}uplication}.
\newblock PLoS Genet. 2011 Oct;7(10):e1002337.

\end{thebibliography}



\end{document}

